src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -627,16 +606,18 @@ private DocumentSnapshot lookupByPathNoXn(
                 .document(encodePathAsFirestoreDocumentName(lookupPath));
 
         RuntimeException lastException = null;
-        for (int retryNum = 0; retryNum < LOOKUP_RETRIES; retryNum++) {
+        for (int retryNum = 0; retryNum < fireStoreUtils.getFirestoreRetries(); retryNum++) {
             logger.info("FirestoreDirectoryDao lookupByPathNoXn - iteration {}", retryNum);
             try {
                 ApiFuture<DocumentSnapshot> docSnapFuture = docRef.get();
                 return docSnapFuture.get();
             } catch (AbortedException | ExecutionException ex) {
                 lastException = fireStoreUtils.handleExecutionException(ex, "lookupByPathNoXn");
             }
-            TimeUnit.SECONDS.sleep(LOOKUP_WAIT_SECONDS);
+            final long retryWait = SLEEP_BASE_MILLISECONDS * Double.valueOf(Math.pow(2.5, retryNum)).longValue();
+            TimeUnit.SECONDS.sleep(retryWait);
```

Comment:
Is this a bug? Your sleep base duration is milliseconds but this is using seconds.
```suggestion
            TimeUnit.MILLISECONDS.sleep(retryWait);
```

Reply:
Yes, this is a bug! Thank you!